doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -2,7 +2,615 @@
+
+1. Select the **My First Kaa Application** application and click **Generate SDK**.
+
+    ![SDK generation](attach/generate_sdk.png)
+1. Click **Add SDK profile**.
+
+    ![Add SDK](attach/add_sdk.png)
+1. In the **Add SDK profile** window, fill in the fields as shown in the following
+screenshot  and then click **Add**.
+
+    ![Configure SDK](attach/configure_sdk.png)
+1. Now the SDK is configured and ready to be generated. Click **Generate SDK** for
+corresponding SDK profile. In the **Generate SDK** window select the target platform for
+your SDK and click Generate SDK.
+
+    ![Generate SDK](attach/generate_configured_sdk.png)
+
+    ![Generate SDK](attach/select_target_platform.png)
+
+After the SDK is generated, you will be presented with a window asking you to save a
+.jar file with the generated SDK (for Java) or an archive with the generated SDK (for C,
+C++ or Objective-C). Specify the file name and location on your computer and then click
+**Save**. The SDK is now downloaded to your computer.
+
+Note that in this example we are generating the SDK based on the default profile and
+notification schemas. These schemas are automatically populated during the creation of
+the application.
+If necessary, you can overwrite them using [Admin UI]({{root_url}}/Administration-guide/Tenants-and-applications-management/).
+
+# Sample client application
+
+Once you have downloaded the SDK, you can use it in your sample project. The following
+code block illustrates a simple desktop application that will send virtual temperature
+data from the Kaa endpoint with required configuration.
+
+NOTE: After generating the C++/Objective-C SDKs, you need to build them before creating the application.
+
+## Preparation
+
+<ul class="nav nav-tabs">
+  <li class="active"><a data-toggle="tab" href="#prep-c">C SDK</a></li>
+  <li><a data-toggle="tab" href="#prep-java">Java SDK</a></li>
+</ul>
+
+<div class="tab-content">
+
+<div id="prep-c" class="tab-pane fade in active" markdown="1" >
+
+Before you start with C application code, CMake file must be filled.
+Unpack C SDK to the `kaa` directory and create `CMakeLists.txt` file in the application dir.
+
+```cmake
+cmake_minimum_required(VERSION 2.8.8)
+project(kaa-application C)
+
+find_package(OpenSSL REQUIRED)
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -g -Wall -Wextra")
+
+add_subdirectory(kaa)
+
+add_executable(kaa-app main.c)
+target_link_libraries(kaa-app kaac crypto)
+
+```
+
+Main source file will be almost empty for now.
+
+```c
+int main(void)
+{
+
+}
+```
+
+To validate that build system works as expected, create dummy `main.c` file and
+trigger a build.
+
+```
+mkdir build
+cd build
+cmake ..
+make
+```
+
+If all is done right, you will see demo application executable in your directory.
+
+```
+$ ls -l kaa-app
+-rwxr-xr-x 1 user 53944 Jun 10 12:36 kaa-app
+```
+
+</div>
+
+<div id="prep-java" class="tab-pane fade" markdown="1" >
+
+Describe preparation steps before using Java SDK
```

Comment:
"Describe preparation steps before using Java SDK" ?

Such a stub should not be left in the documentation.